### DISCUSSION TITLE
Error in text fields.

### DISCUSSION THREAD

**Discussion by mdgt21** [filamentphp/filament#14052] [Help]
### Package

Form builder

### Package Version

v3.0.0.0

### How can we help you?

When I try to limit my TextFields they won't let me do it, I can enter as many characters as I want and it still won't limit. This is my code.

` TextInput::make('codigo')
                            ->minLength(6)
                            ->maxLength(6)
                            ->required()
                            ->numeric()
                            ->label('Código del Cupón')
                            ->unique(Cupon::class, ignoreRecord: true)                          
                            ]),`
                            
                            
![image](https://github.com/user-attachments/assets/69564520-c96e-455a-a465-6adca0fedfac)

**Comment by leandrocfe** [comment]
I believe the min/max methods are effective only when working with `text` data types, not `numeric` ones.
Try mask

```php
TextInput::make('codigo')->mask('999999')
```

**Reply by mdgt21** [reply]
It has worked in terms of the code and the value is more than 4 digits, but in other cases where I need only 3 it does not work, an example is in a percentage of offer, because in this specific case I need this to go from 1 to 100, but if I put mask this only allows me to enter the number 100, this is my code    

`Forms\Components\TextInput::make('descuento_porcentaje')
                            ->label('Descuento en Porcentaje')
                            ->mask(100)
                            ->suffix('%')
                            ->required(fn($get) => $get('tipo_descuento') === 'porcentaje')
                            ->hidden(fn($get) => $get('tipo_descuento') !== 'porcentaje')
                            ->regex('/^\d{1,2}(\.\d{1,2})?$/')
                            ->validationMessages([
                                'regex' => 'El descuento debe tener como máximo dos dígitos enteros y dos decimales.',
                            ]),`
First image with mask 
![Captura de pantalla 2024-08-26 101259](https://github.com/user-attachments/assets/3aca010f-f7de-4643-9473-4ac069ac397a)

Without mask
![Captura de pantalla 2024-08-26 101354](https://github.com/user-attachments/assets/81e8614d-eae7-4ba7-90ec-ffdef1645f59)

**Reply by leandrocfe** [reply]
_I need this to go from 1 to 100..._

```php
TextInput::make('descuento_porcentaje')
->numeric()
->minValue(1)
->maxValue(100)
```

**Reply by mdgt21** [reply]
I tried that and it is still the same. The TextInput is not limited in any way, neither with mask, nor with minValue, nor with Maxvalue. The same thing happened to me with markdown

```
Forms\Components\TextInput::make('descuento_porcentaje')
                            ->label('Descuento en Porcentaje')
                            ->numeric()
                            ->step('0.01')
                            ->minValue(1)
                            ->maxValue(100)
                            ->suffix('%')
                            ->required(fn($get) => $get('tipo_descuento') === 'porcentaje')
                            ->hidden(fn($get) => $get('tipo_descuento') !== 'porcentaje')
                          
```

**Reply by leandrocfe** [reply]
Please explain what you are trying to do

**Reply by mdgt21** [reply]
Ok, I have a form in which I enter several data for products, these products can have a percentage offer, the problem is that if I put minValue, maxValue, minLenght, maxLenght the input does not limit as it should, what I want to do is that the percentage is not greater than 100, but with filament I have the problem that in an input I can always put values higher than 100, and what I want is to have the validation on the client-side.
